### PR TITLE
Refactors status displays

### PR DESCRIPTION
**Pull request by AffectedArc07**
## What Does This PR Do
Refactors status displays to use their own setter proc and global list instead of the current awful radio signal system.

Part of #18685 

As an added benefit, status displays will now instantly respond when you click the button in the PDA or comms console, and they will work instantly without you needing to blank it first.

## Why It's Good For The Game
The old signal system must be purged, and this is the way forward.

## Testing
- Tested all buttons on the comms console
- Tested alert level swapping
- Tested all buttons on the PDA
- Tested supply displays

## Changelog
:cl: AffectedArc07
add: Status displays now update instantly if you change them with the console or PDA
/:cl:

### CODE REVIEW

File: code/game/machinery/ai_display.dm [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+				if("Dorfy")
+					new_display = "ai_urist"
+				if("Facepalm")
+					new_display = "ai_facepalm"
+				if("Friend Computer")
+					new_display = "ai_friend"
+
+		// BSOD
+		if(AI_DISPLAY_MODE_BSOD)
+			new_display ="ai_bsod"
```
Comment (Sirryan2002) [inline code/game/machinery/ai_display.dm:109]:
```suggestion
			new_display = "ai_bsod"
```

File: code/game/machinery/status_display.dm [hunk 2]
```diff
@@ -18,49 +14,34 @@
...
+	/// icon_state of alert picture
+	var/picture_state
+	/// Are we spooked?
+	var/spookymode = FALSE
+	/// Line 1 of a custom message, if any
+	var/message1
+	/// Line 2 of a custom message, if any
+	var/message2
+	/// Is this a supply display?
+	var/is_supply = FALSE
```
Comment (Sirryan2002) [inline code/game/machinery/status_display.dm:32]: shouldn't this just be a subtype? (for future reference that's all, otherwise this would be a map edit)

**Review by Sirryan2002 — COMMENTED** [review]
While signals aren't my forte, everything looks pretty good, just a small thing.

File: code/game/machinery/status_display.dm (hunk 2 above)
Comment (AffectedArc07) [reply code/game/machinery/status_display.dm:32]: The supply subtype defaults this to true.

Dont quote me on this, but I am pretty sure `GetVar()` instruction calls are cheaper than `istype()`.

**Review by hal9000PR — APPROVED** [review]

**Pull request merged**